src/github.com/getlantern/balancer/balancer.go

```diff
@@ -102,11 +102,17 @@ func (b *Balancer) Dial(network, addr string) (net.Conn, error) {
 		trustedOnly = true
 	}
 
+	var lastDialer *dialer
 	for i := 0; i < dialAttempts; i++ {
 		d, pickErr := b.pickDialer(trustedOnly)
 		if pickErr != nil {
 			return nil, pickErr
 		}
+		if d == lastDialer {
```

Comment:
Like whenever there's a failure, we kind of reset and dial all simultaneously. Whichever connects first is the new default dialer.

I know that's a totally different approach, but TBH I'm not a huge fan of the weighted balancer stuff because it means we'll always sometimes try to connect to a potentially really slow proxy. If that's a key CSS file or something, it will completely slow down the page loading (could actually be a reason why users say pro is slow).

I basically don't think the concept of a "balancer" is really what we want.